app/lib/three_scale/analytics/user_tracking.rb

```diff
@@ -31,11 +31,19 @@ def method_missing(method_sym, *args, &block)
         end
 
         def segment_configured?
-          defined?(::Segment)
+          Features::SegmentConfig.enabled? && defined?(::Segment)
         end
 
         def initialize(config = {})
-          @adapter = segment_configured? ? ::Segment::Analytics.new(config) : NullAdapter.new
+          @config = config
+        end
+
+        private
+
+        attr_reader :config
+
+        def adapter
+          segment_configured? ? ::Segment::Analytics.new(config) : NullAdapter.new
```

Comment:
I removed the memoization of this to make the tests pass but I am not sure if it is a good idea or it will make it considerably slower 🤔 the actual root cause of being hard to test and working without this memoization is this line:
https://github.com/3scale/porta/blob/06948bbd7118be15b77e561e269fa6b4878fb2bf/app/lib/three_scale/analytics/user_tracking.rb#L43

Any reviewer should give an opinion and ideas 😝

Reply:
Why are the test failing?

Reply:
Response by Carlos (in case that anyone reads this in the future):

> let supose that the first test that runs was the one with segment enabled. The `Segment` constant assigned to the class will be the `::Segment::Analytics.new(config)` adapter
> and we can’t test the `non segment` after it.. because the constant was already assigned